ui/src/shared/components/StaticLegend.js

```diff
@@ -0,0 +1,120 @@
+import React, {PropTypes, Component} from 'react'
+import _ from 'lodash'
+import uuid from 'node-uuid'
+
+const style = {
+  position: 'absolute',
+  width: 'calc(100% - 32px)',
+  bottom: '8px',
+  left: '16px',
+  paddingTop: '8px',
+}
+
+const removeMeasurement = (label = '') => {
+  const [measurement] = label.match(/^(.*)[.]/g) || ['']
+  return label.replace(measurement, '')
+}
+
+const staticLegendItemClassname = (visibilities, i, hoverEnabled) => {
+  if (visibilities.length) {
+    return `${hoverEnabled
+      ? 'static-legend--item'
+      : 'static-legend--single'}${visibilities[i] ? '' : ' disabled'}`
+  }
+
+  // all series are visible to match expected initial state
+  return 'static-legend--item'
+}
+
+class StaticLegend extends Component {
+  constructor(props) {
```

Comment:
Instead of
```
constructor(props) {
  super(props)

  this.state = {
    visibilities: [],
  }
}
```

We can start doing
```
class StaticLegend extends Component {
  state = {
    visibilities: []
  }
  
  ...
```

Reply:
adding this to style guide